specifyweb/frontend/js_src/lib/components/Interactions/PrepDialog.tsx

```diff
@@ -63,14 +64,20 @@ export function PrepDialog({
     return mutatedPreparations as RA<PreparationData>;
   }, [rawPreparations, itemCollection]);
 
-  const [selected, setSelected] = useLiveState<RA<number>>(
-    React.useCallback(
-      () => Array.from({ length: preparations.length }).fill(0),
-      [preparations.length]
-    )
+  // Change to use an object for selected state allowing null values
+  const [selected, setSelected] = useLiveState<{
+    [key: string]: number | null;
+  }>(
+    React.useCallback(() => {
+      return preparations.reduce((acc, preparation) => {
```

Comment:
a bit more type safe:

```suggestion
      return preparations.reduce<{ [key: string]: number | null }>((acc, preparation) => {
        acc[preparation.preparationId] = null;
        return acc;
      }, {});
```

and we have a utility type for this as it's used very often:

```suggestion
      return preparations.reduce<R<number | null>>((acc, preparation) => {
        acc[preparation.preparationId] = null;
        return acc;
      }, {});
```